### DISCUSSION TITLE
Binding Problem in Oracle SQL Query

### DISCUSSION THREAD

**Discussion by Sagestum** [Budibase/budibase#13898] [Help]
Hello,

I have an issue with a binding in an Oracle SQL query.

I'm trying to create an IN clause to search for multiple SR numbers.

This works:
```sql
WHERE t.ticketno IN ('SR-000456546', 'SR-000456422')
```

However, when I use it as a binding, I get an empty response:
```sql
WHERE t.ticketno IN ({{ticketno}})
```
![image](https://github.com/Budibase/budibase/assets/11883823/39e51dcf-1f1a-4800-9492-30062d19fa1e)

Could you help me understand why this is happening and how to resolve it?

Thank you!

**Comment by Sagestum** [comment]
By the way, in MSSQL it looks like this works just fine.

**Comment by mike12345567** [comment]
Hi @Sagestum - you can use the environment variable `SQL_LOGGING_ENABLE=1` to see what queries Budibase is actually performing when you run the query - Budibase doesn't just replace the binding in the query with the input string, it uses prepared statements in SQL queries to protect against SQL injection. In Oracle DB it may not be possible to do a multi-parameter IN query with prepared statements.

https://docs.budibase.com/docs/hosting-settings

**Reply by Sagestum** [reply]
I have looked at it, and while the queries look different from what I'm used to. But the query that doesn't work doesn't look different from the one that works.

![image](https://github.com/Budibase/budibase/assets/11883823/aa1d1250-5eb9-441c-baf7-c14a3f2aba8d)


First Line: 
![image](https://github.com/Budibase/budibase/assets/11883823/0925432c-ffe3-4b21-b448-d1b0dc59b6d5)

Second Line: 
![image](https://github.com/Budibase/budibase/assets/11883823/c3409b05-f88e-43e9-afd1-46e12daf0118)

Third Line: 
![image](https://github.com/Budibase/budibase/assets/11883823/d01ee8a4-4647-404f-9dcc-225c76c10654)

The third would be the required one, but it gives an empty response. Is this a bug?

**Reply by mike12345567** [reply]
Hi @Sagestum - can you try running a query like what you see printed out of the third one directly? This might give some more explanation of what is going wrong, something like the `WHERE t.ticketno IN ( :1 ) values=....` - if you can run queries like this directly against your Oracle DB it might tell us why the `WHERE t.ticket IN ( :1 )` syntax doesn't work.
